server/sonar-server/src/main/java/org/sonar/server/rule/RuleCreator.java

```diff
@@ -151,11 +147,15 @@ private static void validateRuleKey(List<String> errors, String ruleKey) {
     }
   }
 
-  private Optional<RuleDefinitionDto> loadRule(RuleKey ruleKey, DbSession dbSession) {
+  private Optional<RuleDefinitionDto> loadRule(DbSession dbSession, int ruleId) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused private "loadRule" method. ([squid:UnusedPrivateMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2F7.1%2FSONAR-10313%2Fremove_key_from_rule_indexes&issues=AWFL-Jp4NbtQu1yJ8qeD&open=AWFL-Jp4NbtQu1yJ8qeD)

[ISSUE_KEY]: AWFL-Jp4NbtQu1yJ8qeD